### PR TITLE
fix(ui): Fix fetching of 3 worst metrics web vital transactions [INGEST-752]

### PR DESCRIPTION
**Pull request by matejminar**
image

### Problem
To get the three worst transactions with respect to a web vital, we have to count the number of transactions per transaction name that were classified as "poor" and rank them by that count in descending order.

This means that we cannot use `list` query for both the transactions list and vital bar anymore. It would've been easy to add one more `vitalBar` request to `Queries` object and call it a day but I wanted to figure this out without increasing the number of API requests (2). 

### Solution
Now the `list` query is used only for the transactions list and with `chart` query we are doing a bit more data massaging but we can use it for both the main chart and vital bar.

As a nice by-product of this change, I was able to get rid of `transformMetricsToVitalList` and reuse `transformMetricsToList`.

### CODE REVIEW

File: static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx [hunk 1]
```diff
@@ -104,19 +104,20 @@ export function VitalWidgetMetrics(props: PerformanceWidgetProps) {
             project={project}
             environment={environment}
             query={new MutableSearch(eventView.query)
-              .addFilterValues('transaction', [
-                widgetData.list.data[selectedListIndex].transaction,
-              ])
+              .addFilterValues(
+                'transaction',
+                widgetData.list.data.map(listItem => listItem.transaction) as string[]
```
Comment (jjbayer) [inline static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:109]: I tried this with `?datasource=snuba` and it looks like this translates to the query
```
transaction:/api/{:orgId}/zap transaction:/api/{:orgId}/foo transaction:/api/{:orgId}/bar
```
which is equivalent to 
```
transaction:/api/{:orgId}/zap AND transaction:/api/{:orgId}/foo AND transaction:/api/{:orgId}/bar
```

Can we change this to `transaction IN [...]`?

Comment (jjbayer) [reply static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:109]: `?datasource=mock` basically ignores the query, which is why it did not catch this.

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Review by jjbayer — APPROVED** [review]
Looks good now, the new syntax translates to snuba condition
```python
Condition(
    Column('tags[transaction]'),
    Op.IN, 
    ['/api/{:orgId}/bar', '/api/{:orgId}/zap', '/api/{:orgId}/foo']
)
```

Before:
image

After:
image

**Pull request merged**